### ISSUE TITLE
File content end part is not showing on plain mode

### ISSUE THREAD

**Issue by dalisoft** [sharkdp/bat#1426]
**What version of `bat` are you using?**
`bat 0.17.1`

**Describe the bug you encountered:**
<img width="1931" alt="image" src="https://user-images.githubusercontent.com/3511344/101402691-e5e40100-38f5-11eb-9abc-682acd34bd43.png">

**What did you expect to happen instead?**
<img width="1929" alt="image" src="https://user-images.githubusercontent.com/3511344/101402759-001ddf00-38f6-11eb-8e18-295704e05e65.png">

**How did you install `bat`?**
`brew install bat`

**Workaround**
Passing `--wrap=character` as argument, but i want it should be by default and like it's performance bit slower

---

- OS: macOS Catalina 10.15.7
- Terminal: Apple Terminal
- Prompt: fish + starship
- Package manager: fish

Executing `https://github.com/sharkdp/bat/blob/master/diagnostics/info.sh` failed, says 
```
❯ ./info.sh
```

system
------

**$ uname -srm**
Darwin 19.6.0 x86_64  

**$ sw_vers**
ProductName:	Mac OS X  
ProductVersion:	10.15.7  
BuildVersion:	19H15  

bat
---

**$ bat --version**
bat 0.17.1  

**$ env**
BAT_THEME=Monokai Extended Bright  

bat_config
----------

bat_wrapper
-----------

No wrapper script for 'bat'.

bat_wrapper_function
--------------------

No wrapper function for 'bat'.

**$ /usr/local/bin/fish --login -i -c functions cat**
```
# Defined in - @ line 1
function cat --wraps=bat --description 'alias cat=bat'
  bat  $argv;
end
```

tool
----

**$ less --version**
less 487 (POSIX regular expressions)  

~/Downloads took 22s 
❯ 

```

**Comment by sharkdp** [comment]
> Executing `https://github.com/sharkdp/bat/blob/master/diagnostics/info.sh` failed, says
> 
> ```
> ❯ sh info.sh
> info.sh: line 7: syntax error near unexpected token `newline'
> info.sh: line 7: `<!DOCTYPE html>'
> ```

Looks like you downloaded the HTML page. Please try
```
wget https://raw.githubusercontent.com/sharkdp/bat/master/diagnostics/info.sh
```

**Comment by sharkdp** [comment]
Ok, I think I see what is going on here.

First, `-p` enables `--wrap=never`. This is by design (see #289). Second, we recently enabled another feature which passes `-S` ("chop long lines") to `less` if `--wrap=never` is set.

In combination, `-S` is also passed to `less` if `-p`/`--plain` is used. I think it's pretty clear from your example that this is not ideal. `.pub` files have a single long line and using `bat -p` on them, we should be able to see the full content.

For now, you can work around the issue by using `-p` twice (`bat -pp …`). This also disables the pager.

@eth-p @Lazza @gahag What do you think?

**Comment by eth-p** [comment]
That's a fun, unexpected behavior. Maybe we should only pass `-S` if `--wrap=never` is explicitly set?

Considering my use case for `-p`/`--p` (copying text), it wouldn't make sense to have `less` page long lines.

**Comment by Lazza** [comment]
It seems from the first screenshot that bat is invoking `less` (which does not close itself) even if the file is a single line. Is this expected?

**Comment by eth-p** [comment]
> It seems from the first screenshot that `bat` is invoking `less` (which does not close itself) even if the file is a single line. Is this expected?

It depends. If BAT_PAGER is set or if `--pager` is specified manually or in the config file, bat won't provide default arguments to `less`.

Among the default arguments, `--quit-if-one-screen` is one of them.

**Comment by dalisoft** [comment]
@sharkdp Thanks for reply. I updated first comment with `info.sh` data

**Comment by dalisoft** [comment]
@sharkdp @eth-p I used `-pp` and seems it works, thanks for tip. I didn't knew about this

**Comment by sharkdp** [comment]
Let's keep this open for now

**Comment by gahag** [comment]
Well, this is really unfortunate. I can't think of a better workaround right now, so I'll agree with what @eth-p suggested:
> Maybe we should only pass -S if --wrap=never is explicitly set?

**Comment by sharkdp** [comment]
closed via #1486

**Comment by sharkdp** [comment]
Bugfix released in [bat v0.18](https://github.com/sharkdp/bat/releases/tag/v0.18.0).
